### PR TITLE
added in function to check number of atoms in xyz text

### PR DESCRIPTION
**Pull request by adewyer**

### CODE REVIEW

File: src/basis/basis.js [hunk 1]
```diff
@@ -7,6 +7,12 @@ import { ATOMIC_COORD_UNITS, HASH_TOLERANCE } from "../constants";
 import { Lattice, nonPeriodicLatticeScalingFactor } from "../lattice/lattice";
 import math from "../math";
 
+
+/**
+ * Maximum number of atoms allowed in materials designer basis or a file upload basis.
```
Comment (timurbazhirov) [inline src/basis/basis.js:12]: This message is against the encapsulation of concepts. A person using this library has to know about materials designer and file upload to understand this. A better approach would be to call this something independent of the other packages, like `maximumNumberAtomsSmallMolecule`. Then explain the logic that everything beyond that is to be considered a "large" molecule and thus subject to different processing.

Comment (timurbazhirov) [reply src/basis/basis.js:12]: Alternatively, we can just include `nonPeriodicMaxAtomCount` inside the `Basis` class as a static variable

Comment (adewyer) [reply src/basis/basis.js:12]: changed to static variable.

File: src/parsers/xyz.js [hunk 2]
```diff
@@ -33,16 +33,41 @@ function validateLine(xyzLine, index) {
...
-        .trim()
-        .lines()
-        .filter((x) => x.trim() !== "")
-        .forEach(validateLine);
+    try {
+        s(xyzTxt)
+            .trim()
+            .lines()
+            .filter((x) => x.trim() !== "")
+            .forEach(validateLine);
```
Comment (timurbazhirov) [inline src/parsers/xyz.js:61]: Let's rename previous `validate` to `validateFormat` and then use it in lines 57-61 inside `validateAll`

File: tests/parsers/xyz.js [hunk 3]
```diff
@@ -11,4 +14,12 @@ describe("Parsers:XYZ", () => {
             "units",
         ]);
     });
+    it("should return error number for a structure having less than the maximum number of atoms", () => {
+        const error = validate(smallBasis);
+        expect(error).to.be.equal(0);
+    });
+    it("should return the error number for a structure having more than the maximum number of atoms", () => {
+        const error = validate(largeBasis);
+        expect(error).to.be.equal(2001);
```
Comment (timurbazhirov) [inline tests/parsers/xyz.js:23]: Should also check for 1001 error

File: src/parsers/xyz.js [hunk 4]
```diff
@@ -34,15 +34,52 @@ function validateLine(xyzLine, index) {
...
+    } catch (err) {
+        return 1001;
+    }
+    return 0;
```
Comment (timurbazhirov) [inline src/parsers/xyz.js:66]: The logic for dealing with error codes should be in validateAll.

This function should just throw an Error like validateNumberOfAtoms above

**Review by timurbazhirov — APPROVED** [review]

**Pull request merged**